### PR TITLE
chp5

### PR DESCRIPTION
**Pull request by shmuhammadd**
Chapter5 presentation

### CODE REVIEW

File: 05-workflow.Rmd [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+    -   Learn how to write **self-contained reprexes**
+
+    
+<!-- your comment 
+Writing reprexes, self-contained chunks of code that illustrate a problem. Reprexes are a powerful debugging technique, and they are essential if you want to get help from someone else.
+ --> 
+
+       
+       
+  ![](images/workflow.png)
```
Comment (russHyde) [inline 05-workflow.Rmd]: Could you upload the figures, please.
But put the files in `./images/05-workflow/`
Then make sure that your image-links are rewritten to point to `./images/05-workflow/workflow.png` etc

File: 05-workflow.Rmd [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+##  Debugging 
+
+
+- something will go wrong definitely. 
+
+- it takes years of experience to write code that works the first time (So, we need a robust workflow for identifying and fixing mistakes)
+
+- Specific focus to three debugging challenges to Shiny apps
+
+  ![](images/workflow.png)
```
Comment (russHyde) [inline 05-workflow.Rmd]: Again, please rewrite the image path

File: 05-workflow.Rmd [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+<!-- - R’s debugging tools include the traceback, browser, debug, debugonce, trace, and recover functions -->
+
+<!-- - it takes years of experience in any language before you can reliably write code that works the first time. -->
+
+<!-- - So,bugs do happens. -->
+
+<!-- - Three main cases of problems: -->
+
+
+  ![](images/debugging_challenges.png)
```
Comment (russHyde) [inline 05-workflow.Rmd]: Again, please change to `./images/05-workflow/debugging_challenges.png`

File: 05-workflow.Rmd [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+
+- If you cant debug the error, it is time to ask for help at Shiny Community by creating Reprex
+
+- A reprex is just some R code that works when you copy and paste it into a R session on another computer
+
+- Good reprex, makes it easy for others to help you debug your app 
+
+- Below is an example of Shiny reprex
+
+  ![](images/workflow.png)
```
Comment (russHyde) [inline 05-workflow.Rmd]: Please change to `./images/05-workflow/workflow.png`

**Review by russHyde — CHANGES_REQUESTED** [review]
Thanks for this. Could you push the requested files as `./images/05-workflow/*.png` and rewrite the relevant paths in your chapter. I'll check if the chapter builds locally once the files are on github.

**Comment by shmuhammadd** [comment]
@russHyde, please sorry for the long silence. I updated the chapter as requested.

File: 05-workflow.Rmd (hunk 2 above)
Comment (shmuhammadd) [reply 05-workflow.Rmd]: Done

**Review by russHyde — COMMENTED** [review]
Looks good. There is a figure "development_cycle" that was added to the repo but wasn't used. Is there a missing image import in the .Rmd?

**Comment by russHyde** [comment]
I'll merge it, but if you want to remove the unused figure, feel free to do that in another PR.

**Review by russHyde — APPROVED** [review]
LGTM (the unused figure isn't an issue)

**Comment by shmuhammadd** [comment]
Ok.

**Pull request merged**